src/OrbitGl/TimeGraph.cpp

```diff
@@ -545,11 +545,25 @@ void TimeGraph::DoUpdateLayout() {
 
   time_window_us_ = max_time_us_ - min_time_us_;
 
-  // Update position and size of children.
-  track_container_->SetPos(GetPos()[0], GetPos()[1]);
-  // TrackContainer height is the free space of the screen.
-  track_container_->SetHeight(GetHeight() - timeline_ui_->GetHeight());
-  timeline_ui_->SetPos(GetPos()[0], GetPos()[1] + track_container_->GetHeight());
+  UpdateChildrenPosAndSize();
+}
+
+void TimeGraph::UpdateChildrenPosAndSize() {
+  // Special case: TimeGraph will set TrackContainer height based on its free space.
+  float total_height_without_track_container = 0;
+  for (orbit_gl::CaptureViewElement* children : GetNonHiddenChildren()) {
+    if (children != track_container_.get()) {
+      total_height_without_track_container += children->GetHeight();
+    }
+  }
+  track_container_->SetHeight(GetHeight() - total_height_without_track_container);
+
+  // Update position of visible children.
+  float current_pos_y = GetPos()[1];
+  for (orbit_gl::CaptureViewElement* children : GetNonHiddenChildren()) {
```

Comment:
Nit: The variable should be called `child`, it's referring to a single child since you're iterating.